src/platform_impl/macos/app_delegate.rs

```diff
@@ -36,10 +48,26 @@ lazy_static! {
             activation_hack::mouse_moved as extern "C" fn(&Object, Sel, id),
         );
 
+        decl.add_method(
+            sel!(application:openFile:),
+            application_open_file as extern "C" fn(&Object, Sel, id, id) -> BOOL,
+        );
+        decl.add_method(
+            sel!(application:openFiles:),
+            application_open_files as extern "C" fn(&Object, Sel, id, id),
+        );
+
```

Comment:
Did you verify that the default behaviour, when not implementing these on the delegate, matches the behaviour we have now with `event_loop.set_file_open_callback(None)`?

Reply:
I can confirm that it does *not* match.

When these are not implemented, the following message is shown when attempting to open a file.

![winit_open - no delegate function defined](https://user-images.githubusercontent.com/8320264/107281645-35f85480-6a5a-11eb-8d45-1598eb9e0d4c.png)

This message isn't shown when the functions are implemented on the delegate, regardless of whether the functions indicate success or failure. I'm not sure how or if we want to deal with this. An argument of leaving it as it is, is that the `Info.plist` file must declare that the application supports opening certain file types for this to work anyway. IIRC the application cannot be selected from the "open with" list in case the `Info.plist` doesn't declare a matching file type.

Reply:
It's probably more correct/expected to preserve the old behavior, though I imagine that'd be quite difficult to do? I don't think this is too big a deal either way.

Maybe it'd be helpful for `open_files` to emit a log message when it's triggered but the callback is `None`?

Reply:
In my latest commit I changed the implementation such that the default behavior (i.e. when the callback is set to `None`) is identical to not having the delegate methods implemented.